File: bench/iteration-recursion/4.go
package main

import (
	"os"
	"strconv"
	"fmt"
	"time"
)

// Recursion function
func fib(n uint64) uint64 {
	if n <= 1 {
		return n
	}

	return fib(n-1) + fib(n-2)
}

func main() {
	var n uint64 = 40
	var args = os.Args[1:]
	if len(args) >= 1 {
		n, _ = strconv.ParseUint(args[0], 10, 64)
	}

	var start = time.Now()

	n = fib(n)

	println(n)

	fmt.Println(time.Now().Sub(start).Nanoseconds())
}

File: bench/iteration-recursion/4.cpp
#include <iostream>
#include <chrono>

using namespace std;

// Recursion function
int fib(unsigned int n)
{
	if (n <= 1)
	{
		return n;
	}

	return fib(n - 1) + fib(n - 2);
}

int main(int argc, char const *argv[])
{
	unsigned int n = 40;
	if (argc == 2)
	{
		n = atoi(argv[1]);
	}

	auto start = chrono::steady_clock::now();

	n = fib(n);

	cout << n << endl;

	cout << chrono::duration_cast<chrono::nanoseconds>(chrono::steady_clock::now() - start).count() << endl;

	return 0;
}
